Loan_Default_Classification_Code_Review_V1.py

```diff
@@ -450,33 +456,13 @@ def histogram_alt(feature):
     #msg = "%s: %f" % (name, roc_auc_score)
 
 
-# In[41]:
-
-
-# prepare configuration for cross validation test harness
-seed = 7
-
-# Compare accuracy score among different models
-results = []
-names = []
-scoring = 'accuracy'
-for name, model in models:
-	kfold = model_selection.KFold(n_splits=10, random_state=seed)
-	cv_results = model_selection.cross_val_score(model, X_train, y_train, cv=kfold, scoring=scoring)
-	results.append(cv_results)
-	names.append(name)
-	msg = "%s: %f (%f)" % (name, cv_results.mean(), cv_results.std())
-	print(msg)
-    
-# boxplot algorithm comparison
-fig = plt.figure()
-fig.suptitle('Algorithm Accuracy Comparison')
-ax = fig.add_subplot(111)
-plt.boxplot(results)
-ax.set_xticklabels(names)
-plt.show()
-
-
+"""
+Interpret this part!
+What is the ROC AUC score?
+Which algorithms are better?
+Why does the performance of some of the algorithms look similar?
```

Comment:
This is a great question to ask me and I will try my best to answer :) I know I might get stuck as I explain them.

LR and LDA had almost same level of performance in this case. LR and LDA have no difference in model function but assumptions on feature distribution and the estimation  of  the  coefficients. The assumption of LDA is that the features need to be normally distributed. (Question though - how to measure if a categorical data is normally distributed? All features need to be normally distributed to make LDA outperform than LR?). "Even though LDA is a constantly better method when the normality assumptions are met, the differences between the methods become negligible with a sample size of 50 and more, when the methods differently allocate only about 0.5% of the cases."   

SVM (kernel = "linear") has closely roc auc score compared to LR. My guess is that 
Questions:
How LDA differs from SVM? In terms of create a